pkg/cloudprovider/aws.go

```diff
@@ -0,0 +1,49 @@
+package cloudprovider
+
+import (
+	"github.com/sirupsen/logrus"
+
+	"github.com/libopenstorage/cloudops"
+	awsops "github.com/libopenstorage/cloudops/aws"
+)
+
+type aws struct {
+	defaultProvider
+
+	ops cloudops.Ops
+}
+
+func (a *aws) init() error {
+	// Do parent init first
+	if err := a.defaultProvider.init(); err != nil {
+		return err
+	}
+	if a.distributionName != eksDistribution {
+		return nil
+	}
+
+	// Initialize aws client for cloud drive permission checks
+	client, err := awsops.NewClient()
+	if err != nil {
+		return err
+	}
+	a.ops = client
+	return nil
+}
+
+func (a *aws) Name() string {
+	return string(cloudops.AWS)
+}
+
+func (a *aws) CheckCloudDrivePermission() error {
+	// TODO: check the permission here by doing dummy drive operations
+	logrus.Infof("JLIAO: doing aws permisison check")
```

Comment:
[nit] Remove before merge, maybe replace with `return fmt.Errorf("CheckCloudDrivePermission not implemented")` just for this PR